working-groups/query-standardization/dsl-survey/ppl/README.md

```diff
@@ -0,0 +1,256 @@
+* **Supported Character Sets**:
+    *  Typically UTF-8.
+* **Dimensional Time Series ID Components**:
+    * Metric names can be separate or a dimension; dimensions can have multiple values (Also See API access to Prometheus ).
+* **Dimension Names, Requirements, and Types**:
+    * Supported based on how they are defined in the data model.
+* **Regular Expressions**:
+    * Supported
+* **Extraction Groups**:
+    * Depending on the data and query structure, they may be supported.
+
+*Keep in mind that OpenSearch PPL provides a querying interface, and some of the behaviors related to data modeling, aggregation, interpolation, and handling missing data might be influenced more by how the data is indexed, stored, and managed within OpenSearch rather than the PPL itself.*
+*Information might vary depending on the specific version or configuration of OpenSearch being used.*
+* * *
+**Logs**
+
+* **Incoming Log Event Model:**
+    * Typically key-value maps with information like hostname, log level, timestamp, and message field.
+    * Structured logs (e.g., JSON) are also common via [Simple Schema for Observability](https://github.com/opensearch-project/opensearch-catalog/tree/main/docs/schema/observability/logs) .
+* **Multiline Log Entries:**
+    * Can be handled, and behavior may vary depending on the configuration and specific requirements.
+* **Special Handling for Stack Traces:**
+    * No specific handling within PPL, but OpenSearch can index stack traces, and searching can be performed on specific lines or content.
+    * Using [Simple Schema for Observability](https://github.com/opensearch-project/opensearch-catalog/tree/main/docs/schema/observability/logs) stack Trace has a structured field.
+* **Field Extraction:**
+    * Automatic extraction from structured log messages like JSON is supported, and [query-time field extraction](https://github.com/opensearch-project/sql/issues/359) is  possible.
+* **Including Surrounding Lines in Results:**
+    *  Not directly supported in PPL (supported within OpenSearch Dashboards ).
+* **Full Text Searching**:
+    *  Supported, including proximity, phrase, fuzzy queries, etc.
+* **Regular Expressions:**
+    * Supported
+* **Extraction Groups:**
+    *  Can be implemented using regular expressions and other mechanisms.
+* **Generating Metrics from Logs:**
+    * Supported via appropriate queries and aggregations.
+* **Basic Graphs Generation:**
+    *  Possible through aggregations and visualizations in tools like OpenSearch Dashboards.
+* **Writing Extraction Queries:**
+    *  Supported; fields can be parsed from logs and aggregations applied.
+* **Aggregations Supported**:
+    *  Sum, min, max, count by, percentiles, etc.
+* **Handling Wrong Types or Missing Values**:
+    *  Behavior can vary; consult OpenSearch documentation for details.
```

Comment:
Could you provide an example or two please? In particular, does the DSL define any particular behavior or does it let the data models bubble up errors?